pkgs/applications/gis/qgis/default.nix

```diff
@@ -1,23 +1,31 @@
 { stdenv, fetchurl, fetchpatch, gdal, cmake, qt4, flex, bison, proj, geos, xlibsWrapper, sqlite, gsl
 , qwt, fcgi, python2Packages, libspatialindex, libspatialite, qscintilla, postgresql, makeWrapper
-, qjson, qca2, txt2tags, openssl
+, qjson, qca2, txt2tags, openssl, darwin, pkgconfig
 , withGrass ? false, grass
 }:
 
 stdenv.mkDerivation rec {
   name = "qgis-2.18.16";
 
   buildInputs = [ gdal qt4 flex openssl bison proj geos xlibsWrapper sqlite gsl qwt qscintilla
-    fcgi libspatialindex libspatialite postgresql qjson qca2 txt2tags ] ++
+    fcgi libspatialindex libspatialite postgresql qjson qca2 txt2tags pkgconfig
+    (stdenv.lib.optional stdenv.isDarwin darwin.apple_sdk.frameworks.IOKit)
+    (stdenv.lib.optional stdenv.isDarwin darwin.apple_sdk.frameworks.ApplicationServices)
```

Comment:
Maybe append `lib.optionals stdenv.isDarwin (with darwin.apple_sdk.frameworks; [IOKit ApplicationServices])` instead of having these individual entries

Reply:
Extending the callPackage scope in `all-packages.nix` with frameworks is the most common approach.